### PR TITLE
Fix state prep operation decomposition with `LightningQubit`

### PR DESCRIPTION
**Pull request by mudit2812**
**Context:**
Bunch of test failures in CI are related to `lightning.qubit` not working with adjoint diff whenever there are state prep operations present, even then they're not supposed to be trainable.

**Description of the Change:**
Update the adjoint diff validation for `lightning.qubit` to decompose state prep operations.

**Benefits:**
Better support for adjoint with `lightning.qubit`.

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -47,6 +47,9 @@
 * Fix random `coverage xml` CI issues.
   [(#635)](https://github.com/PennyLaneAI/pennylane-lightning/pull/635)
 
+* `lightning.qubit` correctly decomposed state preparation operations with adjoint differentiation.
+  [(#)]()
```
Comment (maliasadi) [inline .github/CHANGELOG.md:51]: :eyes:

**Review by maliasadi — APPROVED** [review]
:raised_hands:

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -47,6 +47,9 @@
 * Fix random `coverage xml` CI issues.
   [(#635)](https://github.com/PennyLaneAI/pennylane-lightning/pull/635)
 
+* `lightning.qubit` correctly decomposed state preparation operations with adjoint differentiation.
```
Comment (maliasadi) [inline .github/CHANGELOG.md:50]:
```suggestion
* Skip the decomposition of state preparation operations with adjoint differentiation in `lightning.qubit`.
```

File: .github/CHANGELOG.md (hunk 1 above)
Comment (maliasadi) [reply .github/CHANGELOG.md:51]:
```suggestion
  [(#661)](https://github.com/PennyLaneAI/pennylane-lightning/pull/661)
```

**Comment by maliasadi** [comment]
This is related to PR https://github.com/PennyLaneAI/pennylane/pull/5434.

**Review by vincentmr — APPROVED** [review]
LGTM, feel free to merge when the CI is green.

**Pull request merged**